src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpThurloeDAO.scala

```diff
@@ -36,4 +34,64 @@ class HttpThurloeDAO ( implicit val system: ActorSystem, implicit val executionC
     }
   }
 
+  override def getProfile(userInfo: UserInfo): Future[Option[Profile]] = {
+    val pipeline = addFireCloudCredentials ~> addCredentials(userInfo.accessToken) ~> sendReceive
+    pipeline(Get(UserService.remoteGetAllURL.format(userInfo.getUniqueId))) map { response =>
+      response.status match {
+        case StatusCodes.OK => Option(Profile(unmarshal[ProfileWrapper].apply(response)))
```

Comment:
Doug's concern was that you could get Some(null) in that case, assuming I understood and implemented this correctly.